cmd/kubeadm/app/master/manifests.go

```diff
@@ -355,7 +355,9 @@ func getAPIServerCommand(cfg *kubeadmapi.MasterConfiguration, selfHosted bool, k
 		defaultArguments["proxy-client-cert-file"] = filepath.Join(cfg.CertificatesDir, kubeadmconstants.FrontProxyClientCertName)
 		defaultArguments["proxy-client-key-file"] = filepath.Join(cfg.CertificatesDir, kubeadmconstants.FrontProxyClientKeyName)
 	}
-	if k8sVersion.AtLeast(kubeadmconstants.MinimumNodeAuthorizerVersion) {
+	// TODO: That we have to exclude v1.8.0-alpha.0 here is based on the way kubernetes branches work
+	// v1.7.0-beta.0 == v1.8.0.alpha.0 but they are sorted/compared differently
+	if k8sVersion.AtLeast(kubeadmconstants.MinimumNodeAuthorizerVersion) && cfg.KubernetesVersion != "v1.8.0-alpha.0" {
```

Comment:
I'm not sure that I fully grok what happened here, but I think some new unit tests would be a great way to demonstrate what the expected behavior is (as well as prove that this works how we think it does).

Is the problem that this feature is going to be removed in 1.8, or is there actually a problem with how we do semver and we think that 1.8.0-alpha.0 came before 1.7.0? My interpretation of this PR is the latter, which is actually pretty scary if true (and something we should excalate / fix project-wide, I think).